File: chrome/android/java/src/org/chromium/chrome/browser/permissions/PermissionSettingsBridge.java
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.permissions;

import org.chromium.base.annotations.NativeMethods;
import org.chromium.chrome.browser.profiles.Profile;

/**
 * Utility class that interacts with native to retrieve and set permission-related settings.
 */
public class PermissionSettingsBridge {
    public static boolean shouldShowNotificationsPromo() {
        return PermissionSettingsBridgeJni.get().shouldShowNotificationsPromo(getProfile());
    }

    public static void didShowNotificationsPromo() {
        PermissionSettingsBridgeJni.get().didShowNotificationsPromo(getProfile());
    }

    private static Profile getProfile() {
        return Profile.getLastUsedProfile().getOriginalProfile();
    }

    @NativeMethods
    public interface Natives {
        boolean shouldShowNotificationsPromo(Profile profile);
        void didShowNotificationsPromo(Profile profile);
    }
}

File: chrome/browser/android/permissions/permission_settings_bridge.cc
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/android/scoped_java_ref.h"
#include "chrome/android/chrome_jni_headers/PermissionSettingsBridge_jni.h"
#include "chrome/browser/permissions/adaptive_notification_permission_ui_selector.h"
#include "chrome/browser/profiles/profile_android.h"

using base::android::JavaParamRef;

static jboolean JNI_PermissionSettingsBridge_ShouldShowNotificationsPromo(
    JNIEnv* env,
    const JavaParamRef<jobject>& jprofile) {
  return AdaptiveNotificationPermissionUiSelector::GetForProfile(
             ProfileAndroid::FromProfileAndroid(jprofile))
      ->ShouldShowPromo();
}

static void JNI_PermissionSettingsBridge_DidShowNotificationsPromo(
    JNIEnv* env,
    const JavaParamRef<jobject>& jprofile) {
  AdaptiveNotificationPermissionUiSelector::GetForProfile(
      ProfileAndroid::FromProfileAndroid(jprofile))
      ->PromoWasShown();
}
